### PR TITLE
Run installer on uki livecd mode and use the new init detection

### PR DESCRIPTION
**Pull request by Itxaka**

### CODE REVIEW

File: packages/static/kairos-overlay-files/files/system/oem/02_agent.yaml [hunk 1]
```diff
@@ -1,11 +1,11 @@
 name: "Start agent"
 stages:
   boot:
-    - if: '[ ! -f "/run/cos/recovery_mode" ] && [ -d "/usr/share/systemd" ]'
+    - if: '[ ! -f "/run/cos/recovery_mode" ] && [ "$(kairos-agent state get kairos.init)" == "systemd" ]'
```
Comment (mauromorales) [inline packages/static/kairos-overlay-files/files/system/oem/02_agent.yaml:4]: not for now, but seeing this, makes me feel like we could simply have some querying mechanism in the agent e.g.

```
kairos-agent state is kairos.init.systemd
echo $?
1
```

File: packages/static/kairos-overlay-files/files/system/oem/52_installer.yaml [hunk 2]
```diff
@@ -31,11 +43,14 @@ stages:
         - echo "tty1::respawn:/usr/bin/kairos-agent interactive-install --shell tty1" >> /etc/inittab
   boot:
     - if: |
-          [ "$(kairos-agent state get boot)" == "livecd_boot" ] && [ -f "/sbin/rc-service" ]
+        [ "$(kairos-agent state get boot)" == "livecd_boot" ] && [ -f "/sbin/rc-service" ]
```
Comment (mauromorales) [inline packages/static/kairos-overlay-files/files/system/oem/52_installer.yaml:46]: should there not be a 

```
"$(kairos-agent state get kairos.init)" == "openrc"
```

here instead?

Comment (Itxaka) [reply packages/static/kairos-overlay-files/files/system/oem/52_installer.yaml:46]: mmhm yeah, good point. Becuase openrc is "simpler" to check I forgot about using the state xD

Comment (Itxaka) [reply packages/static/kairos-overlay-files/files/system/oem/52_installer.yaml]: done!

File: packages/static/kairos-overlay-files/files/system/oem/02_agent.yaml (hunk 1 above)
Comment (Itxaka) [reply packages/static/kairos-overlay-files/files/system/oem/02_agent.yaml:4]: umm, not sure. State is state right? do we want to introduce that kind of operations onto it? would be cool indeed and we could piggyback the existing state I guess, but seems like a can of worms hahaha

**Review by mauromorales — APPROVED** [review]
LGTM

**Pull request merged**